amalgamation.sh

```diff
@@ -73,10 +73,15 @@ function dofile()
     do
         if [[ "${line}" == '#include "'*'"'* ]]; then
             file=$(echo $line| cut -d'"' -f 2)
+            # inlcude all from src/${AMAL_C} except ${AMAL_H}
+            if [ "${file}" == "${AMAL_H}" ] && [ "${RELFILE}" == "src/${AMAL_C}" ]; then
```

Comment:
Oh, this is so that #include "simdjson.h" happens at the beginning of simdjson.cpp, in all cases?

How did anything ever work?? I assume it's because each of the .cpp files includes their own header files?

If that's the case, perhaps the right thing to do is remove #include "simdjson.h" from the top of simdjson.cpp?

Either way, this is a super good catch!

Reply:
So, the reason it all works: 
amalgamation first builds simdjson.h, and adds all the headers to the list of processed headers, and simdjson.cpp is built, all these header from list of processed are skipped including simdjson.h. As, I've mentioned in ["Do not inline file IO functions"]("https://github.com/simdjson/simdjson/pull/741) PR, I wanted to add `_CRT_SECURE_NO_WARNINGS`, which must be at the top of simdjson.cpp before any other headers, and because of the way amalgamation was done, it forces simdjson.h to be at the top. This PR addresses it, and makes sure that amalgamated simdjson.cpp is identical to the original.